uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -216,7 +223,40 @@ define([
        * @return {Boolean} true is user can upload more else false
        */
       this.canUploadMore = function () {
-        return (this.request.files.length + this.request.fileUploader.queue.length) < sharedSettings.fileUploader.queueLimit;
+        var filesWithoutSoftDelete = _.filter(this.request.files, function (file) {
+          return !file.toBeDeleted;
+        });
+
+        return (filesWithoutSoftDelete.length + this.request.fileUploader.queue.length) < sharedSettings.fileUploader.queueLimit;
+      };
+
+      /**
+       * Checks if user can see file list in files tab. It needs to check for
+       * soft delete flag on leave request as well for files.
+       *
+       * @return {Boolean} true is user can see file list
+       */
+      this.canShowFileList = function () {
```

Comment:
There are 2 methods

1. `canUploadMore` checks if user can upload more given the constraint of shared settings (of 10 files) useful to show the warning or info message
2. `canShowFileList ` which only checks if whitespace above loader needs to be shown or not 

i think both are necessary as they have different purpose but if they are combined then maybe we will have to add another variable to pass the `sharedSettings.fileUploader.queueLimit to template`, which is better?